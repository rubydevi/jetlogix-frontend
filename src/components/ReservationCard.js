import React from 'react';
import PropTypes from 'prop-types';

import {
  CustomCard,
  CardContent,
  CardTitle,
  CardText,
} from './styledcomponents/ReservationStyles';

const ReservationCard = ({ aeroplane }) => (
  <div className="col-lg-6 mb-4">
    <CustomCard className="card">
      <CardTitle className="card-title">Reservation Details</CardTitle>
      <CardContent>
        <CardText className="card-text">
          <span>
            <strong>Name:</strong>
            {' '}
            <small>{aeroplane.name}</small>
          </span>
        </CardText>
        <CardText className="card-text">
          <span>
            <strong>Date:</strong>
            {' '}
            <small>{aeroplane.reserved_date}</small>
          </span>
        </CardText>
        <CardText className="card-text">
          <span>
            <strong>Start Time:</strong>
            {' '}
            <small>{aeroplane.start_time}</small>
          </span>
        </CardText>
        <CardText className="card-text">
          <span>
            <strong>End Time:</strong>
            {' '}
            <small>{aeroplane.end_time}</small>
          </span>
        </CardText>
        <CardText className="card-text">
          <span>
            <strong>Start Location:</strong>
            {' '}
            <small>{aeroplane.start_location}</small>
          </span>
        </CardText>
        <CardText className="card-text">
          <span>
            <strong>Destination:</strong>
            {' '}
            <small>{aeroplane.destination}</small>
          </span>
        </CardText>
      </CardContent>
    </CustomCard>
  </div>
);

ReservationCard.propTypes = {
  aeroplane: PropTypes.shape({
    name: PropTypes.string.isRequired,
    reserved_date: PropTypes.string.isRequired,
    start_time: PropTypes.string.isRequired,
    end_time: PropTypes.string.isRequired,
    start_location: PropTypes.string.isRequired,
    destination: PropTypes.string.isRequired,
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  }).isRequired,
};

export default ReservationCard;
