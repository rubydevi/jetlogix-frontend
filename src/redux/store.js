import { configureStore } from '@reduxjs/toolkit';
import aeroplanesReducer from './aeroplanes/aeroplanesSlice';
import reservationsReducer from './reservations/reservationSlice';

const store = configureStore({
  reducer: {
    aeroplanes: aeroplanesReducer,
    reservations: reservationsReducer,
  },
});

export default store;
