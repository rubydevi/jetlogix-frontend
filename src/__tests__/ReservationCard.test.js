import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

import ReservationCard from '../components/ReservationCard';

test('renders ReservationCard component correctly', () => {
  const aeroplane = {
    name: 'Dragon 1',
    reserved_date: '2023-9-9',
    start_time: '10:00',
    end_time: '11:00',
    start_location: 'Lagos',
    destination: 'Abuja',
    id: 1,
  };

  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <ReservationCard aeroplane={aeroplane} />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
