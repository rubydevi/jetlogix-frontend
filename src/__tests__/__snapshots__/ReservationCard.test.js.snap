// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ReservationCard component correctly 1`] = `
<div>
  <div
    class="col-lg-6 mb-4"
  >
    <div
      class="sc-bdnyFh dFAyrg card"
    >
      <h2
        class="sc-dlniIP iLvzBS card-title"
      >
        Reservation Details
      </h2>
      <div
        class="sc-gtsqUy jFmrNm"
      >
        <p
          class="sc-hKFymg cAQZBV card-text"
        >
          <span>
            <strong>
              Name:
            </strong>
             
            <small>
              Dragon 1
            </small>
          </span>
        </p>
        <p
          class="sc-hKFymg cAQZBV card-text"
        >
          <span>
            <strong>
              Date:
            </strong>
             
            <small>
              2023-9-9
            </small>
          </span>
        </p>
        <p
          class="sc-hKFymg cAQZBV card-text"
        >
          <span>
            <strong>
              Start Time:
            </strong>
             
            <small>
              10:00
            </small>
          </span>
        </p>
        <p
          class="sc-hKFymg cAQZBV card-text"
        >
          <span>
            <strong>
              End Time:
            </strong>
             
            <small>
              11:00
            </small>
          </span>
        </p>
        <p
          class="sc-hKFymg cAQZBV card-text"
        >
          <span>
            <strong>
              Start Location:
            </strong>
             
            <small>
              Lagos
            </small>
          </span>
        </p>
        <p
          class="sc-hKFymg cAQZBV card-text"
        >
          <span>
            <strong>
              Destination:
            </strong>
             
            <small>
              Abuja
            </small>
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
